software/displayerang.livecodescript

```diff
@@ -0,0 +1,1402 @@
+               exit repeat -- exit this small repeat test
+            end if
+         end if
+      end repeat
+      if tMoveOn is true then next repeat -- next tLink
+      -- okay, so go ahead and draw the link tLink
+      if pInfo["links"][tLInk]["type"] is not "dive" and pInfo["links"][tLInk]["type"] is not "rise" then -- FLOW
+         put item 1 of pInfo["links"][tLink]["prev"] into tStart
+         put item 1 of pInfo["links"][tLink]["next"] into tEnd
+         put pSizes["node"] / 2 into tClean
+         put (pGUI["nodes"][tStart]["horz"]+tClean) & "," & pGUI["nodes"][tStart]["vert"] into tStartCoord
+         put (pGUI["nodes"][tEnd]["horz"]-tClean) & "," & pGUI["nodes"][tEnd]["vert"] into tEndCoord
+         put tLink into tBox["link"]
+         put tStartCoord & cr & tEndCoord into tBox["line"]
+         put pSizes["link"] into tBox["size"]
+         put pInfo["links"][tLink]["type"] into tBox["type"]
+         put pInfo["links"][tLink]["weight"] into tWeight
+         if tWeight is empty then
+            put 1 into pGUI["links"][tLink]["weight"]
+         else
+            put tWeight into pGUI["links"][tLink]["weight"]
+         end if
+         put tBox["type"] into pGUI["links"][tLink]["type"]
+         put makeLink(tBox) into pGUI["links"][tLink]["graphic"]
+      else if pInfo["links"][tLInk]["type"] is "dive" then -- DIVE 
+         -- what node diving into
+         --- pInfo link>read=#
+         --- pGUI node>#>graphic
+         -- rect of that control
+         -- points = L-10,top cr L,T+10
+         put pInfo["links"][tLink]["read"] into tTheNode
+         put pGUI["nodes"][tTheNode]["graphic"] into tTheNodeName
+         put the rect of tTheNodeName into tTheNodeRect
+         put item 1 of tTheNodeRect into tL
+         put item 2 of tTheNodeRect into tT
+         put (tL-10) & "," & tT & cr & tL & "," & (tT+10) into tStuff["line"]
+         put tLink into tStuff["link"]
+         put 2 into tStuff["size"]
+         put makeScopeLink(tStuff) into pGUI["links"][tLink]["graphic"]
+      else if pInfo["links"][tLInk]["type"] is "rise" then -- RISE
+         -- what node rising out of
+         --- pInfo link>trim=#
+         --- pGUI node>#>graphic
+         -- rect of that control 
+         -- points = right,T+10 cr R+10,top
+         put pInfo["links"][tLink]["bulk"] into tTheNode
+         put pGUI["nodes"][tTheNode]["graphic"] into tTheNodeName
+         put the rect of tTheNodeName into tTheNodeRect
+         put item 3 of tTheNodeRect into tR
+         put item 2 of tTheNodeRect into tT
+         put tR  & "," & (tT+10) & cr & (tR+10) & "," & tT into tStuff["line"]
+         put tLink into tStuff["link"]
+         put 2 into tStuff["size"]
+         put makeScopeLink(tStuff) into pGUI["links"][tLink]["graphic"]
+      end if
+   end repeat
+   -- draw scope links...also crudly -------------------
+   
+   -- make sure the node graphics are on top
+   repeat for each key tNode in pGUI["nodes"]
```

Comment:
@matthewmaier you could use groups to do this layer management by creating a group for nodes and a group for links and another for the preview graphics. Once in the correct order you can create/delete objects inside them and maintain the appropriate layer order.